tests/integration/test_spanner_vector_store.py

```diff
@@ -0,0 +1,558 @@
+                TableColumn(name="title", type="STRING(1024)"),
+                TableColumn(name="metadata_json_column", type="JSON"),
+            ],
+            primary_key="product_name, title, product_id",
+        )
+
+
+class TestStaticUtilityPGSQL:
+    @pytest.fixture(autouse=True)
+    def setup_database(self, client, cleanupPGSQL):
+        yield
+
+    def test_init_vector_store_table1(self):
+        SpannerVectorStore.init_vector_store_table(
+            instance_id=instance_id,
+            database_id=pg_database,
+            table_name=table_name,
+            metadata_columns=[
+                TableColumn(name="product_name", type="TEXT", is_null=False),
+                TableColumn(name="title", type="varchar(36)"),
+                TableColumn(name="price", type="bigint"),
+            ],
+        )
+
+    def test_init_vector_store_table2(self):
+        SpannerVectorStore.init_vector_store_table(
+            instance_id=instance_id,
+            database_id=pg_database,
+            table_name=table_name,
+            id_column="custom_id1",
+            content_column="custom_content_id1",
+            embedding_column="custom_embedding_id1",
+            metadata_columns=[
+                TableColumn(name="product_name", type="TEXT", is_null=False),
+                TableColumn(name="title", type="varchar(36)"),
+                TableColumn(name="price", type="bigint"),
+            ],
+        )
+
+    def test_init_vector_store_table3(self):
+        SpannerVectorStore.init_vector_store_table(
+            instance_id=instance_id,
+            database_id=pg_database,
+            table_name=table_name,
+            id_column=TableColumn(name="product_id", type="varchar(36)", is_null=False),
+            embedding_column=TableColumn(
+                name="custom_embedding_id1", type="float8[]", is_null=True
+            ),
+            metadata_columns=[
+                TableColumn(name="product_name", type="TEXT", is_null=False),
+                TableColumn(name="title", type="varchar(36)"),
+                TableColumn(name="price", type="bigint"),
+            ],
+        )
+
+    def test_init_vector_store_table4(self):
+        SpannerVectorStore.init_vector_store_table(
+            instance_id=instance_id,
+            database_id=pg_database,
+            table_name=table_name,
+            id_column=TableColumn(name="product_id", type="varchar(36)", is_null=False),
+            embedding_column=TableColumn(
+                name="custom_embedding_id1", type="float8[]", is_null=True
+            ),
+            metadata_columns=[
+                TableColumn(name="product_name", type="TEXT", is_null=False),
+                TableColumn(name="title", type="varchar(36)"),
+                TableColumn(name="price", type="bigint"),
+            ],
+            primary_key="product_name, title, product_id",
+        )
+
+
+class TestSpannerVectorStoreGoogleSQL:
+    @pytest.fixture(scope="class")
```

Comment:
Tests show an error:
ScopeMismatch: You tried to access the function scoped fixture cleanupGSQL with a class scoped request object, involved factories:
tests/integration/test_spanner_vector_store.py:210:  def setup_database(client, cleanupGSQL)
tests/integration/test_spanner_vector_store.py:45:  def cleanupGSQL(client)

This is scoped to class but the other fixtures default to function. You may just need to call the clean up step after the yield of the setup_database method